feat(plugins-doc) update documentation for centreon-plugin-applications-protocol-[cifs,ftp,sftp] plugins (#3269)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/applications-protocol-cifs.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Contenu du pack

### Modèles

Le connecteur de supervision **Protocol CIFS** apporte un modèle d'hôte :

* **App-Protocol-Cifs-custom**

Le connecteur apporte les modèles de service suivants
(classés selon le modèle d'hôte auquel ils sont rattachés) :

<Tabs groupId="sync">
<TabItem value="App-Protocol-Cifs-custom" label="App-Protocol-Cifs-custom">

| Alias      | Modèle de service                   | Description                           |
|:-----------|:------------------------------------|:--------------------------------------|
| Connection | App-Protocol-Cifs-Connection-custom | Contrôle la connexion CIFS à distance |

> Les services listés ci-dessus sont créés automatiquement lorsque le modèle d'hôte **App-Protocol-Cifs-custom** est utilisé.

</TabItem>
<TabItem value="Non rattachés à un modèle d'hôte" label="Non rattachés à un modèle d'hôte">

| Alias       | Modèle de service                    | Description                                                          |
|:------------|:-------------------------------------|:---------------------------------------------------------------------|

> | Files-Count | App-Protocol-Cifs-Files-Count-custom | Contrôle permettant le nombre de fichiers dans un répertoire         |

Reviewer: ```suggestion
| Files-Count | App-Protocol-Cifs-Files-Count-custom | Contrôle permettant de vérifier le nombre de fichiers dans un répertoire         |
```

> Pour superviser votre serveur CIFS, il est possible de réaliser une authentification par : 

Reviewer: ?

3. Appliquez le modèle d'hôte **App-Protocol-Cifs-custom**. Une liste de macros apparaît. Les macros vous permettent de définir comment le connecteur se connectera à la ressource, ainsi que de personnaliser le comportement du connecteur.
4. Renseignez les macros désirées. Attention, certaines macros sont obligatoires.

| Macro                    | Description                                                                                                                              | Default value     | Mandatory   |
|:-------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| PROTOCOLCIFSUSERNAME     | Username                                                                                                                                 |                   |             |
| PROTOCOLCIFSPASSWORD     | Password                                                                                                                                 |                   |             |

> | PROTOCOLCIFSTIMEOUT      | Timeout                                                                                                                                  |                   |             |

Reviewer: ?

Reviewer: @Evan-Adam what timeout? what does the plugin do?

2. Renseignez les macros désirées (par exemple, ajustez les seuils d'alerte). Les macros indiquées ci-dessous comme requises (**Obligatoire**) doivent être renseignées.

<Tabs groupId="sync">
<TabItem value="Connection" label="Connection">

| Macro          | Description                                                                                                                            | Default value                             | Mandatory   |
|:---------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------|:-----------:|
| DIRECTORY      | Set the share directory.                                                                                                               |                                           | X           |

> | CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL You can use the following variables: %{status}, %{message}                | %{message} !~ /authentication succeeded/i |             |

Reviewer: ```suggestion
| CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL. You can use the following variables: %{status}, %{message}                | %{message} !~ /authentication succeeded/i |             |
```

| WARNINGTIME    | Warning threshold in seconds.                                                                                                          |                                           |             |
| CRITICALTIME   | Critical threshold in seconds.                                                                                                         |                                           |             |
| EXTRAOPTIONS   | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                                           |             |

</TabItem>
<TabItem value="Files-Count" label="Files-Count">

| Macro                 | Description                                                                                                                            | Default value     | Mandatory   |
|:----------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | MAXDEPTH              | Number of level of directory to check. 0 mean current directory only.                                                                  | 0                 | X           |

Reviewer: ```suggestion
| MAXDEPTH              | Number of levels of directories to check. 0 means current directory only.                                                                  | 0                 | X           |
```

> | DIRECTORY             | Directory name to check.                                                                                                               |                   | X           |

Reviewer: ```suggestion
| DIRECTORY             | Name of the directory  to check.                                                                                                               |                   | X           |
```

> | FILTERFILE            | Filter files (can be a regexp. Directory in the name).                                                                                 |                   |             |

Reviewer: ?

Reviewer: @Evan-Adam filter files according to what?

| CRITICALFILESDETECTED | Thresholds                                                                                                                             |                   |             |
| EXTRAOPTIONS          | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Files-Date" label="Files-Date">

| Macro             | Description                                                                                                                            | Default value     | Mandatory   |
|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| FILE              | File name to check                                                                                                                     |                   | X           |

> | DIRECTORY         | Check files in the directory (no recursive)                                                                                            |                   | X           |

Reviewer: ```suggestion
| DIRECTORY         | Check files in the directory (no recursion)                                                                                            |                   | X           |
```

| WARNINGMTIMELAST  | Thresholds                                                                                                                             |                   |             |
| CRITICALMTIMELAST | Thresholds                                                                                                                             |                   |             |
| EXTRAOPTIONS      | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Files-Size" label="Files-Size">

| Macro        | Description                                                                                                                            | Default value     | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | MAXDEPTH     | Number of level of directory to check. 0 mean current directory only.                                                                  | 0                 | X           |

Reviewer: ```suggestion
| MAXDEPTH     | Number of levels of directories to check. 0 means current directory only.                                                                  | 0                 | X           |
```

| connection [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/connection.pm)]  | App-Protocol-Cifs-Connection-custom   |
| files [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/files.pm)]            | Not used in this Monitoring Connector |
| files-count [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/filescount.pm)] | App-Protocol-Cifs-Files-Count-custom  |
| files-date [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/filesdate.pm)]   | App-Protocol-Cifs-Files-Date-custom   |
| files-size [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/filessize.pm)]   | App-Protocol-Cifs-Files-Size-custom   |
| scenario [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/apps/protocols/cifs/mode/scenario.pm)]      | App-Protocol-Cifs-Scenario-custom     |

### Options disponibles

#### Options génériques

Les options génériques sont listées ci-dessous :

> | Option | Description |

Reviewer: empty

|:-------|:------------|

#### Options des modes

Les options disponibles pour chaque modèle de services sont listées ci-dessous :

<Tabs groupId="sync">
<TabItem value="Connection" label="Connection">

| Option | Description |
|:-------|:------------|

</TabItem>
<TabItem value="Files-Count" label="Files-Count">

| Option | Description |
|:-------|:------------|

</TabItem>
<TabItem value="Files-Date" label="Files-Date">

| Option | Description |
|:-------|:------------|

</TabItem>
<TabItem value="Files-Size" label="Files-Size">

| Option | Description |
|:-------|:------------|

</TabItem>
<TabItem value="Scenario" label="Scenario">

| Option | Description |
|:-------|:------------|

</TabItem>

> </Tabs>

Reviewer: empty

| EXTRAOPTIONS       | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles). | --verbose              |             |

</TabItem>
</Tabs>

3. [Déployez la configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). Le service apparaît dans la liste des services supervisés, et dans la page **Statut des ressources**. La commande envoyée par le connecteur est indiquée dans le panneau de détails du service : celle-ci montre les valeurs des macros.

## Comment puis-je tester le plugin et que signifient les options des commandes ?

Une fois le plugin installé, vous pouvez tester celui-ci directement en ligne
de commande depuis votre collecteur Centreon en vous connectant avec
l'utilisateur **centreon-engine** (`su - centreon-engine`). Vous pouvez tester
que le connecteur arrive bien à superviser une ressource en utilisant une commande
telle que celle-ci (remplacez les valeurs d'exemple par les vôtres) :

```bash
/usr/lib/centreon/plugins/centreon_protocol_cifs.pl \
	--plugin=apps::protocols::cifs::plugin \
	--mode=scenario \
	--hostname='10.0.0.1' \

> 	--cifs-username='' \
> 	--cifs-password='' \

Reviewer: ```suggestion
        --cifs-username='my-username' \
        --cifs-password='my-password' \
```
